studio/hooks/misc/useProjectSubscription.tsx

```diff
@@ -0,0 +1,35 @@
+import { get } from 'lib/common/fetch'
+import { API_URL } from 'lib/constants'
+import useSWR from 'swr'
+
+interface Data {
+  tier: {
+    name: string
+    prod_id: string
+    key: 'FREE' | 'PRO' | 'PAYG'
+    unit_amount: number
+  }
+  addons: any[]
+  billing: {
+    billing_cycle_anchor: number
+    current_period_end: number
+    current_period_start: number
+  }
+}
+
+interface Handlers {
+  refresh: () => void
+}
+
+const useProjectSubscription = (projectRef: string): [Data | null, Handlers] => {
```

Comment:
seems weird we need a hook for this.
already tagged Hieu in main PR comments about it

Reply:
It's ok for now to use `useProjectSubscription`. As we haven't pulled project tier info on `/projects` yet

However as we haven't landed https://github.com/supabase/infrastructure/pull/4702. This PR has to wait for that